clicommand/secret_batch_read.go

```diff
@@ -0,0 +1,137 @@
+package clicommand
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/buildkite/agent/v3/api"
+	"github.com/buildkite/agent/v3/internal/secrets"
+	"github.com/buildkite/agent/v3/jobapi"
+	"github.com/urfave/cli"
+)
+
+type SecretBatchReadConfig struct {
+	GlobalConfig
+	APIConfig
+
+	Keys          []string `cli:"arg:*"`
+	Format        string   `cli:"format" validate:"required"`
+	Job           string   `cli:"job" validate:"required"`
+	SkipRedaction bool     `cli:"skip-redaction"`
+}
+
+var SecretBatchReadCommand = cli.Command{
+	Name:  "batch-read",
+	Usage: "Concurrently fetch a batch of secrets by their keys",
```

Comment:
Mentioning "concurrently" here is implying something that might not be true in future (e.g. replacing all this with a single request). Maybe just
```suggestion
	Usage: "Fetch a batch of secrets by their keys",
```

Reply:
ooh yep, good call